tq/basic_upload.go

```diff
@@ -112,11 +135,10 @@ func (a *basicUploadAdapter) DoTransfer(ctx interface{}, t *Transfer, cb Progres
 		return errors.NewRetriableError(err)
 	}
 
-	// A status code of 403 likely means that an authentication token for the
-	// upload has expired. This can be safely retried.
-	if res.StatusCode == 403 {
-		err = errors.New("http: received status 403")
-		return errors.NewRetriableError(err)
+	switch res.StatusCode {
+	case 403, // likely an expired auth token
+		503: // service unavailable
```

Comment:
Hm, this is definitely a more pedantic review comment, but I am not sure that I'm convinced of this formatting. I like that it will make diffs easier to read should we add more HTTP status codes that warrant calling `errors.NewRetriableError` (though I'm not sure how likely that is to occur), but feel that having the `503:` case arm at the same level of indentation as the case body is confusing.

What do you think about this as an alternative?

```go
switch res.StatusCode {
case 403, 503: // expired auth token, service unavailable, respectively
        return errors.NewRetriableError(...)
}
```

Reply:
I formatted it this way so that the comments described what each status code means. Though a better option is to probably link to the spot in the docs instead.